index.js

```diff
@@ -190,23 +206,30 @@ class MongoTenant {
   injectApi() {
     let me = this;
 
-    this.schema.statics[this.getAccessorMethod()] = function(tenantId) {
-      if (!me.isEnabled()) {
-        return this.model(this.modelName);
-      }
+    Object.assign(this.schema.statics, {
+      [this.getAccessorMethod()]: function (tenantId) {
+        if (!me.isEnabled()) {
+          return this.model(this.modelName);
+        }
 
-      let modelCache = me._modelCache[this.modelName] || (me._modelCache[this.modelName] = {});
+        let modelCache = me._modelCache[this.modelName] || (me._modelCache[this.modelName] = {});
 
-      // lookup tenant-bound model in cache
-      if (!modelCache[tenantId]) {
-        let Model = this.model(this.modelName);
+        // lookup tenant-bound model in cache
+        if (!modelCache[tenantId]) {
+          let Model = this.model(this.modelName);
 
-        // Cache the tenant bound model class.
-        modelCache[tenantId] = me.createTenantAwareModel(Model, tenantId);
-      }
+          // Cache the tenant bound model class.
+          const boundDb = me.createBoundDb(Model.db, tenantId);
+          modelCache[tenantId] = me.createTenantAwareModel(Model, tenantId, boundDb);
```

Comment:
it feels like the db with tenant context should be created within `createTenantAwareModel`